### ISSUE TITLE
PHPUnit 6.4 complains that "This test did not perform any assertions"

### ISSUE THREAD

**Issue by celtric** [phpspec/prophecy#366]
PHP version: 7.1
PHPUnit version: 6.4.3

Code to reproduce issue: https://github.com/celtric/phpunit-prophecy-issue
Sample error: https://circleci.com/gh/celtric/phpunit-prophecy-issue/6

**Comment by celtric** [comment]
After some initial research, it looks like `$this->bindToObjectProphecy();` should've been called inside `MethodProphecy::shouldHave` just as it's being done in `MethodProphecy::should`, in order to register said method prophecy which is later requested by PHPUnit (https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/TestCase.php#L1182).

By registering the method this way, PHPUnit is able to identify it as an assertion and therefore mark the test as non-risky.

**Comment by majkel89** [comment]
I experience same issue. PHPUnit 6.4.4, Prophecy 1.7.3. For a workaround i have used `self::assertTrue(true)`

**Comment by hugochinchilla** [comment]
Just add this snippet in your `tearDown` method.

```
$this->addToAssertionCount(count($this->prophet->getProphecies()));
```

**Comment by fhuitelec** [comment]
For [Mockery](https://github.com/mockery/mockery/blob/master/library/Mockery/Container.php#L419), in the same fashion @hugochinchilla suggested in the `tearDown`:

```php
$this->addToAssertionCount(
    \Mockery::getContainer()->mockery_getExpectationCount()
);
```

**Comment by kennblvnp** [comment]
@hugochinchilla which file do I insert that code? thanks

**Comment by hugochinchilla** [comment]
@kennblvnp you have to put it in the tearDown method of any affected TestCase, you should create a custom base class and make all your tests inherit from it.

**Comment by kennblvnp** [comment]
Thanks for replying, I already solve my problem! I did a quick setup of laravel dusk and its all working fine.

**Comment by umpirsky** [comment]
You can use `beStrictAboutTestsThatDoNotTestAnything="false"` in `phpunit.xml` or `--report-useless-tests` CLI option.

https://phpunit.de/manual/current/en/risky-tests.html

**Comment by coldic3** [comment]
> Are you using https://github.com/phpspec/prophecy-phpunit ? It already does this properly.

Good point! I’ve completely missed that. The trait in that package resolves all the issues 😅

Thanks!
